Services/CO.CDP.Forms.WebApi/UseCase/UpdateFormSectionAnswersUseCase.cs

```diff
@@ -30,12 +30,30 @@ public async Task<bool> Execute((Guid formId, Guid sectionId, Guid answerSetId,
         }
         else if (sharedConsent.SubmissionState == Persistence.SubmissionState.Submitted)
         {
-            formRepository.ClearTracker();
-            sharedConsent.AnswerSets = sharedConsent.AnswerSets.Where(x => x.Section.Type != Persistence.FormSectionType.Declaration).ToList();
```

Comment:
This line was changing the entity state of AnswerSets of type Declaration to Deleted. That was giving later on a false impression that the result was an update while it was removing these and adding new entries from newly created entities.